services/skus/skus.go

```diff
@@ -123,3 +127,164 @@ func validateHardcodedSku(ctx context.Context, sku string) (bool, error) {
 	valid, ok := skuMap[env][sku]
 	return valid && ok, nil
 }
+
+func newOrderItemReqForSubID(set map[string]model.OrderItemRequestNew, subID string) (model.OrderItemRequestNew, error) {
+	key, err := skuNameByMobileName(subID)
+	if err != nil {
+		return model.OrderItemRequestNew{}, model.ErrInvalidMobileProduct
+	}
+
+	result, ok := set[key]
+	if !ok {
+		return model.OrderItemRequestNew{}, model.ErrInvalidMobileProduct
+	}
+
+	return result, nil
+}
+
+func skuNameByMobileName(subID string) (string, error) {
+	switch subID {
+	case "brave.leo.monthly", "beta.leo.monthly", "nightly.leo.monthly", "braveleo.monthly":
+		return "brave-leo-premium", nil
+	case "brave.leo.yearly", "beta.leo.yearly", "nightly.leo.yearly", "braveleo.yearly":
+		return "brave-leo-premium-year", nil
+	default:
+		return "", model.ErrInvalidMobileProduct
+	}
+}
+
+func newCreateOrderReqNewLeo(ppcfg *premiumPaymentProcConfig, item model.OrderItemRequestNew) model.CreateOrderRequestNew {
+	result := model.CreateOrderRequestNew{
+		// No email.
+		Currency: "USD",
+
+		StripeMetadata: &model.OrderStripeMetadata{
+			SuccessURI: ppcfg.SuccessURI,
+			CancelURI:  ppcfg.CancelURI,
+		},
+		PaymentMethods: []string{"stripe"},
+
+		Items: []model.OrderItemRequestNew{item},
+	}
+
+	return result
+}
+
+type premiumPaymentProcConfig struct {
+	SuccessURI string
```

Comment:
question: do these fields need to be exported or can we make them unexported?

Reply:
No, they currently don't need to be exported. This data is not marshalled/unmarshalled, so there is no need for the fields to be exported.

Fixed.